lib/awful/layout/init.lua.in

```diff
@@ -109,7 +109,15 @@ function layout.arrange(screen)
     p.geometry = capi.screen[screen].geometry
     p.clients = client.tiled(screen)
     p.screen = screen
-    layout.get(screen).arrange(p)
+
+    -- Apply layout geometry
+    local gs = layout.get(screen).arrange(p)
```

Comment:
Please use weak tables for each gs implementations, a race condition is possible and, as explainer in the commit below, I would like this to be stored.